client/components/TestRun/index.jsx

```diff
@@ -90,19 +133,46 @@ class TestRun extends Component {
                 </React.Fragment>
             );
             if (!this.state.runComplete) {
-                content = (
-                    <React.Fragment>
-                        <iframe
-                            src={`/aria-at/${git_hash}/${
-                                tests[this.state.currentTestIndex - 1].file
-                            }?at=${at_key}`}
-                            id="test-iframe"
-                        ></iframe>
-                        <Button variant="primary" onClick={this.nextTest}>
-                            Next test
-                        </Button>
-                    </React.Fragment>
-                );
+                if (!test.result) {
+                    content = (
+                        <React.Fragment>
+                            <iframe
+                                src={`/aria-at/${git_hash}/${
+                                    tests[this.state.currentTestIndex - 1].file
+                                }?at=${at_key}`}
+                                id="test-iframe"
+                                ref={this.testIframe}
+                            ></iframe>
+                            <Button
+                                variant="primary"
+                                onClick={this.handleSkipTestClick}
+                            >
+                                Skip to next test
+                            </Button>
+                            <Button
+                                variant="primary"
+                                onClick={this.handleSaveResultClick}
+                            >
+                                Save results and go to next test
+                            </Button>
+                        </React.Fragment>
+                    );
+                } else {
+                    content = (
+                        <React.Fragment>
+                            <div>
+                                RESULTS EXIST!!!!!! TODO: Make a view of the
+                                existng tests
+                            </div>
+                            <Button
+                                variant="primary"
+                                onClick={this.handleNextTestClick}
```

Comment:
Missing aria-label?

Reply:
these buttons don't need labels because their inner text is descriptive enough! Only when there is additional context about the button based on position do we add aria-label specifically